include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,345 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_distributed_cell_data_transfer_h
+#define dealii_distributed_cell_data_transfer_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/smartpointer.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <boost/range/iterator_range.hpp>
+
+#include <functional>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace parallel
+{
+  namespace distributed
+  {
+    /**
+     * Transfer data that is associated with each cell (like error indicators)
+     * while refining and/or coarsening a distributed grid and handles the
+     * necessary communication.
+     *
+     * This class therefore does for cell-related information what
+     * parallel::distributed::SolutionTransfer does for the values of degrees of
+     * freedom defined on a parallel::distributed::Triangulation.
+     *
+     * This class has been designed to operate on miscellaneous data generated
+     * from current solutions that will be stored in non-distributed Vector
```

Comment:
can we be more general here and support `std::vector<T>` where `T` is serializable? What if each cell contains `T=std::pair<double,double>` or an `T=std::set<unsigned int>`?